### PR TITLE
common/relabel: update validation rules from Prometheus

### LINKED ISSUE #4837: Sync loki.relabel with Promtail 2.8.4

### PR DESCRIPTION
**Pull request by tpaschalis**
#### PR Description
This PR updates the validation rules for all `*.relabel` blocks to be aligned with the validations in [upstream Prometheus](https://github.com/prometheus/prometheus/blob/a807dd16160f0816d5f03bfed72a5321372b0de4/model/relabel/relabel.go#L102-L153)'s UnmarshalYAML method.

#### Which issue(s) this PR fixes
Fixes #4837 

#### Notes to the Reviewer
I don't think there's something else to sync for loki.relabel since it's not _really_ using much of Promtail code

#### PR Checklist

- [ ] CHANGELOG.md updated (N/A)
- [ ] Documentation added (N/A) 
- [ ] Tests updated (N/A)
- [ ] Config converters updated (N?A)

### CODE REVIEW

File: component/common/relabel/relabel.go [hunk 1]
```diff
@@ -146,16 +146,13 @@ func (rc *Config) Validate() error {
 	if rc.Modulus == 0 && rc.Action == HashMod {
 		return fmt.Errorf("relabel configuration for hashmod requires non-zero modulus")
 	}
-	if (rc.Action == Replace || rc.Action == KeepEqual || rc.Action == DropEqual || rc.Action == HashMod || rc.Action == Lowercase || rc.Action == Uppercase) && rc.TargetLabel == "" {
+	if (rc.Action == Replace || rc.Action == HashMod || rc.Action == Lowercase || rc.Action == Uppercase || rc.Action == KeepEqual || rc.Action == DropEqual) && rc.TargetLabel == "" {
```
Comment (tpaschalis) [inline component/common/relabel/relabel.go:149]: Re-ordering so it's easier to scan if the Prometheus code has changed something.

File: component/common/relabel/relabel.go [hunk 2]
```diff
@@ -146,16 +146,13 @@ func (rc *Config) Validate() error {
...
 		return fmt.Errorf("relabel configuration for %s action requires 'target_label' value", rc.Action)
 	}
-	if (rc.Action == KeepEqual || rc.Action == DropEqual) && rc.SourceLabels == nil {
-		return fmt.Errorf("relabel configuration for %s action requires 'source_labels' value", rc.Action)
-	}
```
Comment (tpaschalis) [inline component/common/relabel/relabel.go:154]: This never existed upstream, not sure why I'd added it tbh.

File: component/common/relabel/relabel.go [hunk 3]
```diff
@@ -146,16 +146,13 @@ func (rc *Config) Validate() error {
...
-	if (rc.Action == Replace || rc.Action == Lowercase || rc.Action == Uppercase) && !relabelTarget.MatchString(rc.TargetLabel) {
+	if (rc.Action == Replace || rc.Action == Lowercase || rc.Action == Uppercase || rc.Action == KeepEqual || rc.Action == DropEqual) && !relabelTarget.MatchString(rc.TargetLabel) {
 		return fmt.Errorf("%q is invalid 'target_label' for %s action", rc.TargetLabel, rc.Action)
 	}
-	if (rc.Action == Lowercase || rc.Action == Uppercase) && rc.Replacement != DefaultRelabelConfig.Replacement {
+	if (rc.Action == Lowercase || rc.Action == Uppercase || rc.Action == KeepEqual || rc.Action == DropEqual) && rc.Replacement != DefaultRelabelConfig.Replacement {
```
Comment (tpaschalis) [inline component/common/relabel/relabel.go:155]: We were missing these two for this validation.

**Review by thampiotr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**